src/Compilers/CSharp/Portable/Lowering/LocalRewriter/DelegateCacheContainer.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols;
+
+/// <summary>
+/// This type is synthesized to hold the cached delegates that target static method groups.
+/// </summary>
+internal sealed class DelegateCacheContainer : SynthesizedContainer
+{
+    private readonly Symbol _containingSymbol;
+    private readonly NamedTypeSymbol? _constructedContainer;
```

Comment:
>NamedTypeSymbol? [](http://example.com/codeflow?start=21&length=16)

Should this be non-nullable?

Reply:
We don't need it for non-generic containers.

Reply:
> We don't need it for non-generic containers.

Good point. The fact that we unconditionally call ```Construct``` confused me. What exact implementation is getting called?

Reply:
BTW, we could assign ```this``` to this field in that case and keep it non-nullable.

Reply:
> What exact implementation is getting called?

Sorry I don't understand the question. Implementation of what?

Reply:
> Implementation of what?

Of ```Construct``` in ``_constructedContainer = Construct(ConstructedFromTypeParameters);```.

Reply:
As far as I could remember the Construct and AsMember calls were added after I saw some error message for not calling them, in the original PR several years ago. And I'm on my cell phone right now, so I have no idea about it. Any concerns?

Reply:
> And I'm on my cell phone right now, so I have no idea about it.

Just respond when you can.

Reply:
Sure I'll find time to look into it, likely in weekends. If you have any concerns regarding different implementations of Construct, please let me know, because it's more of an API to me.

Reply:
It's `NamedTypeSymbol.Construct(IEnumerable<TypeSymbol>)`, here https://github.com/dotnet/roslyn/blob/01cbd62809ef685ab679e812fab3f1b09e729380/src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs#L981-L989